common/src/abstractions/state.service.ts

```diff
@@ -1,6 +1,250 @@
+import { BehaviorSubject } from 'rxjs';
+
+import { KdfType } from '../enums/kdfType';
+import { UriMatchType } from '../enums/uriMatchType';
+
+import { CipherData } from '../models/data/cipherData';
+import { CollectionData } from '../models/data/collectionData';
+import { EventData } from '../models/data/eventData';
+import { FolderData } from '../models/data/folderData';
+import { OrganizationData } from '../models/data/organizationData';
+import { PolicyData } from '../models/data/policyData';
+import { ProviderData } from '../models/data/providerData';
+import { SendData } from '../models/data/sendData';
+
+import { Account } from '../models/domain/account';
+import { EncString } from '../models/domain/encString';
+import { GeneratedPasswordHistory } from '../models/domain/generatedPasswordHistory';
+import { Policy } from '../models/domain/policy';
+import { StorageOptions } from '../models/domain/storageOptions';
+import { SymmetricCryptoKey } from '../models/domain/symmetricCryptoKey';
+
+import { CipherView } from '../models/view/cipherView';
+import { CollectionView } from '../models/view/collectionView';
+import { FolderView } from '../models/view/folderView';
+import { SendView } from '../models/view/sendView';
+
 export abstract class StateService {
-    get: <T>(key: string) => Promise<T>;
-    save: (key: string, obj: any) => Promise<any>;
-    remove: (key: string) => Promise<any>;
-    purge: () => Promise<any>;
+    accounts: BehaviorSubject<{ [userId: string]: Account }>;
+
+    setActiveUser: (userId: string) => Promise<void>;
+    addAccount: (account: Account) => Promise<void>;
+    purge: (options?: StorageOptions) => Promise<void>;
+
+    getAccessToken: (options?: StorageOptions) => Promise<string>;
```

Comment:
They were sorted alphabetically because of laziness, but now they are sorted by function AND alphabetically in the abstraction & its implementation. Hopefully we don't need a comment for that and it is just obvious, but I'll add one if you still think we need it.